src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -158,20 +160,23 @@ internal CommitManager(ImmutableArray<char> potentialCommitCharacters, RecentIte
                 var mappedSpan = triggerSnapshotSpan.TranslateTo(subjectBuffer.CurrentSnapshot, SpanTrackingMode.EdgeInclusive);
 
                 edit.Replace(mappedSpan.Span, change.TextChange.NewText);
-                var updatedCurrentSnapshot = edit.Apply();
 
+                // The edit updates the snapshot however other extensions may make changes there.
+                // Therefore, it is safe to consume subjectBuffer.CurrentSnapshot for further calculations.
+                edit.Apply();
+                
                 if (change.NewPosition.HasValue)
                 {
-                    view.TryMoveCaretToAndEnsureVisible(new SnapshotPoint(updatedCurrentSnapshot, change.NewPosition.Value));
+                    view.TryMoveCaretToAndEnsureVisible(new SnapshotPoint(subjectBuffer.CurrentSnapshot, change.NewPosition.Value));
```

Comment:
Shouldn't this also be making a snapshot point against the appropriate snapshot and translating it forward? i.e. the updatedCurrentSnapshot was correct *here* but not further below? (This also presumes TryMoveCaretAndEnsureVisible translates forward instead of crashing.)

Reply:
Let us consider this as a separate issue:  [https://github.com/dotnet/roslyn/issues/31383](https://github.com/dotnet/roslyn/issues/31383)

---
In reply to: [236495952](https://github.com/dotnet/roslyn/pull/29016#discussion_r236495952) [](ancestors = 236495952)